lib/ecto_ldap/adapter.ex

```diff
@@ -295,10 +322,23 @@ defmodule Ecto.Ldap.Adapter do
     split_and_not_nil(t, count - 1, false, [h|acc])
   end
 
+  def update(repo, schema_meta, fields, filters, _autogen_id, returning, options) do
+    case update(schema_meta, fields, filters) do
+      :ok ->
+        {:ok, fields}
+      error ->
+        IO.inspect error
+        {:invalid, []}
+    end
+  end
+
   def load(:id, value), do: {:ok, value}
   def load(:string, value), do: {:ok, convert_from_erlang(value)}
+  def load({:array, :string}, []), do: {:ok, []}
+  def load({:array, :string}, value), do: {:ok, convert_from_erlang(value)}
```

Comment:
I thought that `convert_from_erlang` was already handling that?

def convert_from_erlang(list=[head|_]) when is_list(head), do: Enum.map(list, &convert_from_erlang/1)`